src/Symfony/Component/Cache/Traits/MemcachedTrait.php

```diff
@@ -139,11 +117,34 @@ public static function createConnection($servers, array $options = array())
                 if (isset($params['query'])) {
                     parse_str($params['query'], $query);
                     $params += $query;
+                    $options += $query;
```

Comment:
this means that the precedence is:
1. options provided as 2nd argument
2. win over options provided as part of the DSN
3. who then win over defaults.

But in RedisTrait, 1. and 2. are swapped:
1. options provided as part of the DSN
2. win over options provided as 2nd arg
3. who then win over defaults.

This allows "the infrastructure" to take over the app config.
I think it also makes sense to do the same here.